packages/tools/src/tools/OverlayGridTool.ts

```diff
@@ -0,0 +1,384 @@
+    }
+
+    const { annotations } = this.editData;
+    if (!annotations) {
+      return renderStatus;
+    }
+
+    [...annotations.values()].forEach((annotation) => {
+      const sourceViewport = annotation.data.sourceViewport;
+      if (!sourceViewport) {
+        return renderStatus;
+      }
+
+      if (sourceViewport.id === targetViewport.id) {
+        // If the source viewport is the same as the current viewport, we don't need to render
+        return renderStatus;
+      }
+
+      if (
+        sourceViewport.getFrameOfReferenceUID() !==
+        targetViewport.getFrameOfReferenceUID()
+      ) {
+        return renderStatus;
+      }
+
+      if (!annotation) {
+        return renderStatus;
+      }
+
+      const styleSpecifier: StyleSpecifier = {
+        toolGroupId: this.toolGroupId,
+        toolName: this.getToolName(),
+        viewportId: enabledElement.viewport.id,
+      };
+
+      const { focalPoint, viewPlaneNormal } = targetViewport.getCamera();
+      const { viewPlaneNormal: sourceViewPlaneNormal } =
+        sourceViewport.getCamera();
+
+      if (this.isParallel(viewPlaneNormal, sourceViewPlaneNormal)) {
+        // If the source and target viewports are parallel, we don't need to render
+        return renderStatus;
+      }
+
+      const targetViewportPlane = csUtils.planar.planeEquation(
+        viewPlaneNormal,
+        focalPoint
+      );
+
+      const imageIds = sourceViewport.getImageIds();
+      for (let i = 0; i < imageIds.length; i++) {
+        const { pointSet1, pointSet2 } = this.calculateImageIdPointSets(
+          imageIds[i]
+        );
+
+        let pointSetToUse = pointSet1;
+
+        let topBottomVec = vec3.subtract(
+          vec3.create(),
+          pointSet1[0],
+          pointSet1[1]
+        );
+        topBottomVec = vec3.normalize(
+          vec3.create(),
+          topBottomVec
+        ) as Types.Point3;
+
+        let topRightVec = vec3.subtract(
+          vec3.create(),
+          pointSet1[2],
+          pointSet1[0]
+        );
+        topRightVec = vec3.normalize(vec3.create(), topRightVec);
+
+        const newNormal = vec3.cross(
+          vec3.create(),
+          topBottomVec,
+          topRightVec
+        ) as Types.Point3;
+
+        if (this.isParallel(newNormal, viewPlaneNormal)) {
+          return renderStatus;
+        }
+
+        // check if it is perpendicular to the viewPlaneNormal which means
+        // the line does not intersect the viewPlaneNormal
+        if (this.isPerpendicular(topBottomVec, viewPlaneNormal)) {
+          // 'use pointSet2';
+          pointSetToUse = pointSet2;
+        }
+
+        const lineStartWorld = csUtils.planar.linePlaneIntersection(
+          pointSetToUse[0],
+          pointSetToUse[1],
+          targetViewportPlane
+        );
+
+        const lineEndWorld = csUtils.planar.linePlaneIntersection(
+          pointSetToUse[2],
+          pointSetToUse[3],
+          targetViewportPlane
+        );
```

Comment:
This can be done outside of this renderAnnotation since 
- poinSet is coming from imageIds (not camera zoom pan dependant)
- targetViewportPlane is camera dependant but only focal point and normal

So this means that if we can pre-calculate them on the setToolActive/Enabled here in this loop we only need to convert the `lineStart/EndWorld` to canvas using the targetViewport camera (which has pan/zoom and is important to re-calculate it)